app/assets/javascripts/discourse/app/pre-initializers/inject-discourse-objects.js

```diff
@@ -26,22 +25,69 @@ export function registerObjects(app) {
   app.register("site-settings:main", siteSettings, { instantiate: false });
 }
 
+function injectServiceIntoService({ container, app, property, specifier }) {
+  // app.inject doesn't allow implicit injection of services into services.
```

Comment:
TIL! Is it documented anywhere?

Reply:
I haven't found it in any documentation. But it seems very deliberate:

https://github.com/emberjs/ember.js/blob/36505f1b4296c307e5584ac1104bfdf086b86e99/packages/%40ember/-internals/container/lib/registry.ts#L511

So you can't import services into services, controllers into controllers, etc. I guess the motivation might be to avoid injecting things into themselves? (but… it seems to work fine 🤷‍♂️ )